.github/workflows/resource-indexer.yml

```diff
@@ -0,0 +1,39 @@
+name: rebuild the resource collection index
+
+on:
+  # Run at ~4am UTC time which is (± an hour) 4am UK, 5am Switzerland, midnight
+  # US east coast, 9pm US west coast so that for most users (and most
+  # developers) the index regenerates overnight
+  schedule:
+    - cron: '* 4 * * *'
+
+  # Manually triggered using GitHub's UI
+  workflow_dispatch:
+
+jobs:
+  rebuild-index:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - run: npm ci
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::827581582529:role/GitHubActionsRoleResourceIndexer
+          credential-lifetime: 3600 # seconds, or 1 hour, which is currently the default and should be plenty
+      - name: Rebuild the index
+        run: |
+          node resourceIndexer/main.js \
+            --gzip --output resources.json.gz
+      - name: Make a backup of the existing (current) index so that we can restore if needed
+        run: |
+          aws s3 cp s3://nextstrain-inventories/resources.json.gz s3://nextstrain-inventories/resources.json.gz.previous
```

Comment:
It'd be very easy for two bad runs to not leave us with a backup to restore from. I'd think to do this more robustly by enabling object versioning on the bucket and setting a bucket lifecycle applied to this key prefix which permanently deletes past versions after a fixed time.

Reply:
Thanks - versioning enabled on the bucket, lifecycles adjusted accordingly, backup step removed, and docs updated